### PR TITLE
sample for Distributed counters

### PR DESCRIPTION
**Pull request by laljikanjareeya**
Add sample for Distributed Counters concepts.

Samples for Distributed Counters in other languages:
https://firebase.google.com/docs/firestore/solutions/counters

### CODE REVIEW

File: samples/solution-counters.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+async function main() {
+  // [START create_counter]
+  async function createCounter(docRef, numShards) {
+    const colRef = docRef.collection("shards");
+    for (let count = 0; count < numShards; count++) {
+      const newDoc = colRef.doc(count.toString());
+      await newDoc.set({ count: 0 });
+    }
+  }
+  // [END create_counter]
```
Comment (IlyaFaer) [inline samples/solution-counters.js]: Better write `fs_` prefix in tag names. Well, that's how Ben Whitehead [recommended](https://github.com/GoogleCloudPlatform/golang-samples/pull/959#discussion_r318639468), I know, that this prefix is not "worldwide"

**Review by IlyaFaer — APPROVED** [review]

File: samples/solution-counters.js (hunk 1 above)
Comment (laljikanjareeya) [reply samples/solution-counters.js]: Thanks for your feedback!! in above case all samples are in one repository while here it's firestore specific, so I don't think we need to add prefix.

File: samples/solution-counters.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  // [START increment_counter]
+  async function incrementCounter(docRef, numShards) {
+    const shardId = Math.floor(Math.random() * numShards);
+    const shardRef = docRef.collection("shards").doc(shardId.toString());
+    return shardRef.update({ count: FieldValue.increment(1) });
```
Comment (schmidt-sebastian) [inline samples/solution-counters.js]: If you replaced this with 

```
return shardRef.set({ count: FieldValue.increment(1) }, {merge: true});
```

you would not have to create the counters and initialize them with zero.

File: samples/solution-counters.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    const colRef = docRef.collection("shards");
+    for (let count = 0; count < numShards; count++) {
+      const newDoc = colRef.doc(count.toString());
+      await newDoc.set({ count: 0 });
+    }
+  }
+  // [END create_counter]
+
+  // [START increment_counter]
+  async function incrementCounter(docRef, numShards) {
```
Comment (schmidt-sebastian) [inline samples/solution-counters.js]: This function doesn't use "await" and hence you can drop "async".

File: samples/solution-counters.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    return snapshotList.docs.reduce((count, documentSnapshot) => {
+      return count + documentSnapshot.data().count;
+    }, 0)
+  }
+  // [END get_count]
+
+  // Create a new client
+  const firestore = new Firestore();
+  const docRef = firestore.doc('distributed_counter_samples/distributed_counter');
+  const numberOfshard = 10;
```
Comment (schmidt-sebastian) [inline samples/solution-counters.js]: numberOfShards?

File: samples/test/solution-counters.test.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+'use strict';
+
+const { execSync } = require('child_process');
+const { assert } = require('chai');
+const exec = cmd => execSync(cmd, { encoding: 'utf8' });
+
+describe('distributed counter', () => {
+    it('should create, increase and get counter', () => {
+        const output = exec('node solution-counters.js');
```
Comment (schmidt-sebastian) [inline samples/test/solution-counters.test.js:23]: Hm. Can you rewrite this test to call `incrementCounter()` and `getCount()` directly? That lets you verify the output without string comparisons.

Comment (laljikanjareeya) [reply samples/test/solution-counters.test.js:23]: For nodejs the samples are designed like user can run the file independently without any code changes and can see the output.
all the function will calls the internally, and in test file we are just executing the sample file and matching the output from the sample, the is same in all the repository.

**Pull request closed without merging**